run.sh

```diff
@@ -90,8 +97,41 @@ function die() {
 function lookup_scenario_group() {
     local group="$1"
 
-    activate_venv
-    cat < scenario_groups.yml | python -c 'import yaml; import sys; key = sys.argv[1]; data = sys.stdin.read(); g = yaml.safe_load(data)[key]; [[print(t) for t in s] if isinstance(s, list) else print(s) for s in g]' "${group}"
+    # formatted perl code with -e for readabiliy
+    # shellcheck disable=SC2016
+    cat < scenario_groups.yml \
+        | env GROUP="${group}" perl -0777 \
```

Comment:
> @pawelchcki do we know if this perl setup will be available for all users of system tests? Linux, and OSX laptops + CIs ?

- macOS: yes (perl + YAML module)
- Linux: git depends on perl, but for YAML depends on whether a system dependent need the YAML module
- CI: git depends on perl, but YAML module is absent

Reply:
> @pawelchcki: I was wondering about it thought - why not use Docker itself for these kind of things ? Since we expect Docker to work for everyone wanting to use system-tests. Its our lingua franca - of sorts.
You could embed the conversion script as small docker image. And just call cat yaml | docker run --rm system-tests/scripts process.yaml > output|

I tried this:

```
cat < scenario_groups.yml | docker run --rm alpine:3.16 /bin/sh -c "apk add --quiet --no-progress python3 py3-yaml && python3 -c 'import yaml; import sys; key = sys.argv[1]; data = sys.stdin.read(); g = yaml.safe_load(data)[key]; print(repr(g)); [[print(t) for t in s] if isinstance(s, list) else print(s) for s in g]' 'ONBOARDING_SCENARIOS'"
```

But it does not work:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: 'NoneType' object is not subscriptable
```

I feared as much, as piping into (or out of) docker is a strange affair (there's a whole lot of whacky forwarding involved because containers are run under `dockerd`, not as children of `docker run`)

Anyway I don't think docker is available on macOS in CI.

Reply:
I was thinking of reinstating `activate_venv` (although properly conditioned behind the nix shell check this time) is problematic because when using `./build.sh -d -i runner` + `./run.sh +d` at no point the python dependencies are needed locally, so they may be entirely absent.

That said, the runner image would have these dependencies, so based on `+d` it could either locally `activate_venv` as needed _or_ run in docker (if I manage to achieve that)

Reply:
Alright, I rolled back to using python and moved the conditioned `activate_venv` part earlier so that it's available for the whole `run.sh` script.